File: 2729-check-if-the-number-is-fascinating/2729-check-if-the-number-is-fascinating.go
func isFascinating(n int) bool {
    
    newDigit := strconv.Itoa(n) + strconv.Itoa(2 * n) + strconv.Itoa(3 * n)
    uniqueDigits := make(map[rune]bool)
    
    for _, digit := range newDigit {
        if digit == '0' {
            return false
        }
        
        uniqueDigits[digit] = true
    }
    
    return len(uniqueDigits) == 9 && len(newDigit) == 9
}

File: 2729-check-if-the-number-is-fascinating/2729-check-if-the-number-is-fascinating.py
class Solution:
    def isFascinating(self, n: int) -> bool:
        new_digit = str(n) + str(2 * n) + str(3 * n)
        unique_digits = set(new_digit)
        return (
            len(unique_digits) == 9 and len(new_digit) == 9 and "0" not in unique_digits
        )


# TODO check - https://leetcode.com/contest/biweekly-contest-106/submissions/detail/968072317/
